core/src/main/scala/org/apache/spark/sql/delta/ActiveOptimisticTransactionRule.scala

```diff
@@ -16,25 +16,74 @@
 
 package org.apache.spark.sql.delta
 
+import org.apache.hadoop.fs.Path
 import org.apache.spark.sql.SparkSession
-import org.apache.spark.sql.catalyst.expressions.SubqueryExpression
+import org.apache.spark.sql.catalyst.expressions.{And, Expression, SubqueryExpression}
 import org.apache.spark.sql.catalyst.planning.PhysicalOperation
-import org.apache.spark.sql.catalyst.plans.logical.{LogicalPlan, V2WriteCommand}
+import org.apache.spark.sql.catalyst.plans.logical.{Filter, LogicalPlan, Subquery, SupportsSubquery, V2WriteCommand}
 import org.apache.spark.sql.catalyst.rules.Rule
 import org.apache.spark.sql.delta.files.{PinnedTahoeFileIndex, TahoeLogFileIndex}
+import org.apache.spark.sql.delta.sources.DeltaSQLConf
+import org.apache.spark.sql.execution.datasources.LogicalRelation
 
 /**
- * This rule tries to find if we attempt to read the same table when there is an active transaction.
- * If so, reading the same table should use the same snapshot of the transaction, and the
- * transaction should also record the filters that are used to read the table.
+ * This rule:
+ * - Tries to find if we attempt to read the same table when there is an active transaction. If so,
+ *   reading the same table should use the same snapshot of the transaction, and the transaction
+ *   should also record the filters that are used to read the table.
+ * - Generates and applies partition filters from the data filters for generated columns if
+ *   generated column partition filter optimization is enabled
```

Comment:
Do these need to be part of the same rule? It seems weird to group them together as mostly unrelated

Reply:
I agree that it's a little weird (and could use some renaming). But it helps ensure we use the optimized filters to record read files in an optimistic transaction. However, #966 is removing this file anyways, and we've decided to refactor all of this into [`PrepareDeltaScan`](https://github.com/delta-io/delta/blob/503881673da016ba29b5ca79374819a199a2d613/core/src/main/scala/org/apache/spark/sql/delta/stats/PrepareDeltaScan.scala).